test/components/converters/test_markdown_to_document.py

```diff
@@ -31,17 +31,20 @@ def test_run(self, test_files_path):
             assert "What to build with Haystack" in doc.content
             assert "# git clone https://github.com/deepset-ai/haystack.git" in doc.content
 
-    def test_run_with_meta(self):
+    def test_run_with_meta(self, test_files_path):
         bytestream = ByteStream(data=b"test", meta={"author": "test_author", "language": "en"})
 
         converter = MarkdownToDocument()
 
         with patch("haystack.components.converters.markdown.MarkdownIt"):
-            output = converter.run(sources=[bytestream], meta=[{"language": "it"}])
```

Comment:
Similar to the [other PR](https://github.com/deepset-ai/haystack/pull/6615), we should test all the different input formats. Not only single dict as input for `meta` but also list of dicts. https://github.com/deepset-ai/haystack/pull/6615#pullrequestreview-1793175058

Reply:
I wouldn't re-test all cases though, the `normalize_metadata` function is already tested separately and it would lead to a lot of duplication imho... I'd rather test that the function is being called, by passing a metadata that needs to be normalized to work, and stop there. Wdyt?

Reply:
Okay, yes, fair enough. 👍